### ISSUE TITLE
No error when calling print().format()

### ISSUE THREAD

**Issue by rowlap** [PyCQA/pylint#2200]
This code triggers no pylint errors, although the closing bracket is in the wrong place.
```
def display_errors(filename, errors):
    """docstring"""
    for error in errors:
        print("{}: {}").format(filename, error)
```
However when run, the `print().format()` bug is hit:
```
{}: {}
AttributeError: 'NoneType' object has no attribute 'format'
```
Is this something which could be flagged up by pylint? (string-missing-format? print-returns-None?). It's possible the code really does want to print unmodified `{}` output, but the presence of ".format()" on the same line makes it suspicious.

### pylint --version output
```
No config file found, using default configuration
pylint 1.8.2,
astroid 1.6.1
Python 2.7.14 (default, Sep 16 2017, 17:49:51)
[GCC 7.3.0]
```

**Comment by PCManticore** [comment]
Yes, this is something that pylint should flag, thanks for raising the issue!

**Comment by sushobhit27** [comment]
case 1:
>>> n = astroid.extract_node('"abc:{}".format(123)')
>>> print(n)
Call(func=<Attribute.format l.1 at 0x6fffd71d9e8>,
     args=[<Const.int l.1 at 0x6fffd71da20>],
     keywords=None)
>>> print(n.func.expr)
Const.str(value='abc:{}')

case 2:
>>> s = 'abc:{}'
>>> n = astroid.extract_node(s.format(123))
>>> print(n)
AnnAssign(simple=1,
          target=<AssignName.abc l.1 at 0x6fffd72f0b8>,
          annotation=<Const.int l.1 at 0x6fffd72f080>,
          value=None)
>>>

@PCManticore shouldn't both cases be inferred in same way?

**Comment by PCManticore** [comment]
@sushobhit27 Nope, because you actually have different nodes there:

1. `'"abc:{}".format(123)'` -> this is a string that looks like `"abc:12"`
2. `'abc:{}'.format(123)` -> this is no longer a string, if this would be a file, it will basically contain `abc:123` which is an annotation assign.

**Comment by sushobhit27** [comment]
@PCManticore  got it, but many times I do following thing:
```python
class Employee:
    def __init__(self, name, age, dept):
        self.name = name
        self.age = age
        self.dept = dept

    def __str__(self):
        return 'Name:{}, EmpID: {}, Dept:{}'.format(self.name, self.age, self.dept)

    def to_string(self):
        s = 'Name:{}, EmpID: {}, Dept:{}'
        return s.format(self.name, self.age, self.dept)

emp = Employee('Sushobhit', 786, 'HR')
print(emp)
print(emp.to_string())

Name:Sushobhit, EmpID: 786, Dept:HR
Name:Sushobhit, EmpID: 786, Dept:HR

```
both __str__ and to_string indeed replace value into placeholders in the string, IMO both should be interpreted as string only.
Annotation assignment should be inferred only when the statement is like:
def foo(a: int, b: bool, c: 123, d: 'whatever'): pass

Anyways using current inferrence logic also this can be fixed.

**Comment by PCManticore** [comment]
But that's not a string, if you open a shell in Python 3.6 you can actually do the following:
```
In [3]: a:2 =4

In [4]: a
Out[4]: 4
```
This is exactly what `AnnAssign` is,  an annotation form for assignment targets, we can't interpret it as string when it's not supposed to represent that.
